### PR TITLE
feat(entry admin): Improve the UX on the entry admin panel

### LINKED ISSUE #98: Add a search bar for Entry in Admin

### LINKED ISSUE #99: View Entries by User

### PR DESCRIPTION
**Pull request by samoehlert**
Closes #99 
Closes #98 

I had to branch off of this to make some other changes as well unfortunately just to get a working copy of the stack. Our expiration was not TZ aware which broke any add (from WUI or admin). Also had to fix the translatorConsumer because websockets were broken and spewing errors constantly.

### CODE REVIEW

File: scram/route_manager/models.py [hunk 1]
```diff
@@ -97,7 +99,7 @@ class Entry(models.Model):
     history = HistoricalRecords()
     when = models.DateTimeField(auto_now_add=True)
     who = models.CharField("Username", default="Unknown", max_length=30)
-    expiration = models.DateTimeField(default="9999-12-31 00:00")
+    expiration = models.DateTimeField(default=datetime(9999, 12, 31, 0, 0, tzinfo=timezone.utc))
```
Comment (grigorescu) [inline scram/route_manager/models.py:102]: Looks like you should remove `from django.utils import timezone` and change this to `tzinfo=datetime.timezone.utc`

File: config/consumers.py [hunk 2]
```diff
@@ -36,7 +35,7 @@ async def connect(self):
 
         for route in routes:
             for element in elements:
-                msg = await sync_to_async(partial(element.websocketmessage))
+                msg = await sync_to_async(lambda e: e.websocketmessage)(element)
```
Comment (grigorescu) [inline config/consumers.py:38]: I'm not sure it matters, but curious why we're changing this?

File: scram/route_manager/admin.py [hunk 3]
```diff
@@ -14,7 +14,14 @@ class ActionTypeAdmin(SimpleHistoryAdmin):
     list_display = ("name", "available")
 
 
-admin.site.register(Entry, SimpleHistoryAdmin)
+@admin.register(Entry)
+class EntryAdmin(SimpleHistoryAdmin):
+    """Configure how Entries show up in the Admin site."""
+
+    list_filter = ("is_active", "who")
```
Comment (grigorescu) [inline scram/route_manager/admin.py:21]: I think we should change this to:

```
list_filter = [
    "is_active",
    ("who", admin.RelatedOnlyFieldListFilter),
]
```

That will only return users who have actually blocked a thing (or blocked a thing that meets the other filter criteria).

File: config/consumers.py (hunk 2 above)
Comment (samoehlert) [reply config/consumers.py:38]: TranslatorConsumer was barfing hundreds of logs faster than i could hit ctrl-c about this. I'm not gonna lie I don't 100% understand why it started happening all of a sudden (was on a fresh branch with no code from myself added), but this fixes it. I think something nested in there wasn't being called asynchronously? This was the error that was being fired off at an insanely fast rate if I added an entry.

```python
django-1  | INFO:     connection closed
django-1  | INFO 2024-12-24 16:36:02,399 server 48 274929645376 connection closed
django-1  | INFO 2024-12-24 16:36:02,403 consumers 48 274929645376 Translator connected
django-1  | INFO:     ('172.18.0.7', 34334) - "WebSocket /ws/route_manager/translator_block/" [accepted]
django-1  | INFO 2024-12-24 16:36:02,406 websockets_impl 48 274929645376 ('172.18.0.7', 34334) - "WebSocket /ws/route_manager/translator_block/" [accepted]
django-1  | INFO:     connection open
django-1  | INFO 2024-12-24 16:36:02,409 server 48 274929645376 connection open
django-1  | ERROR:    Exception in ASGI application
django-1  | Traceback (most recent call last):
django-1  |   File "/usr/local/lib/python3.12/site-packages/django/db/models/fields/related_descriptors.py", line 218, in __get__
django-1  |     rel_obj = self.field.get_cached_value(instance)
django-1  |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/django/db/models/fields/mixins.py", line 15, in get_cached_value
django-1  |     return instance._state.fields_cache[cache_name]
django-1  |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^
django-1  | KeyError: 'websocketmessage'
django-1  | 
django-1  | During handling of the above exception, another exception occurred:
django-1  | 
django-1  | Traceback (most recent call last):
django-1  |   File "/usr/local/lib/python3.12/site-packages/uvicorn/protocols/websockets/websockets_impl.py", line 184, in run_asgi
django-1  |     result = await self.app(self.scope, self.asgi_receive, self.asgi_send)
django-1  |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/uvicorn/middleware/proxy_headers.py", line 75, in __call__
django-1  |     return await self.app(scope, receive, send)
django-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/channels/routing.py", line 71, in __call__
django-1  |     return await application(scope, receive, send)
django-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/channels/routing.py", line 150, in __call__
django-1  |     return await application(
django-1  |            ^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/channels/consumer.py", line 94, in app
django-1  |     return await consumer(scope, receive, send)
django-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/channels/consumer.py", line 58, in __call__
django-1  |     await await_many_dispatch(
django-1  |   File "/usr/local/lib/python3.12/site-packages/channels/utils.py", line 51, in await_many_dispatch
django-1  |     await dispatch(result)
django-1  |   File "/usr/local/lib/python3.12/site-packages/channels/consumer.py", line 73, in dispatch
django-1  |     await handler(message)
django-1  |   File "/usr/local/lib/python3.12/site-packages/channels/generic/websocket.py", line 173, in websocket_connect
django-1  |     await self.connect()
django-1  |   File "/app/config/consumers.py", line 38, in connect
django-1  |     msg = await sync_to_async(element.websocketmessage)()
django-1  |                               ^^^^^^^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/django/db/models/fields/related_descriptors.py", line 236, in __get__
django-1  |     rel_obj = self.get_object(instance)
django-1  |               ^^^^^^^^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/django/db/models/fields/related_descriptors.py", line 199, in get_object
django-1  |     return qs.get(self.field.get_reverse_related_filter(instance))
django-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/django/db/models/query.py", line 633, in get
django-1  |     num = len(clone)
django-1  |           ^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/django/db/models/query.py", line 380, in __len__
django-1  |     self._fetch_all()
django-1  |   File "/usr/local/lib/python3.12/site-packages/django/db/models/query.py", line 1881, in _fetch_all
django-1  |     self._result_cache = list(self._iterable_class(self))
django-1  |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/django/db/models/query.py", line 91, in __iter__
django-1  |     results = compiler.execute_sql(
django-1  |               ^^^^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/django/db/models/sql/compiler.py", line 1560, in execute_sql
django-1  |     cursor = self.connection.cursor()
django-1  |              ^^^^^^^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/django/utils/asyncio.py", line 24, in inner
django-1  |     raise SynchronousOnlyOperation(message)
django-1  | django.core.exceptions.SynchronousOnlyOperation: You cannot call this from an async context - use a thread or sync_to_async.
django-1  | ERROR 2024-12-24 16:36:02,421 websockets_impl 48 274929645376 Exception in ASGI application
django-1  | Traceback (most recent call last):
django-1  |   File "/usr/local/lib/python3.12/site-packages/django/db/models/fields/related_descriptors.py", line 218, in __get__
django-1  |     rel_obj = self.field.get_cached_value(instance)
django-1  |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/django/db/models/fields/mixins.py", line 15, in get_cached_value
django-1  |     return instance._state.fields_cache[cache_name]
django-1  |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^
django-1  | KeyError: 'websocketmessage'
django-1  | 
django-1  | During handling of the above exception, another exception occurred:
django-1  | 
django-1  | Traceback (most recent call last):
django-1  |   File "/usr/local/lib/python3.12/site-packages/uvicorn/protocols/websockets/websockets_impl.py", line 184, in run_asgi
django-1  |     result = await self.app(self.scope, self.asgi_receive, self.asgi_send)
django-1  |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/uvicorn/middleware/proxy_headers.py", line 75, in __call__
django-1  |     return await self.app(scope, receive, send)
django-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/channels/routing.py", line 71, in __call__
django-1  |     return await application(scope, receive, send)
django-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/channels/routing.py", line 150, in __call__
django-1  |     return await application(
django-1  |            ^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/channels/consumer.py", line 94, in app
django-1  |     return await consumer(scope, receive, send)
django-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/channels/consumer.py", line 58, in __call__
django-1  |     await await_many_dispatch(
django-1  |   File "/usr/local/lib/python3.12/site-packages/channels/utils.py", line 51, in await_many_dispatch
django-1  |     await dispatch(result)
django-1  |   File "/usr/local/lib/python3.12/site-packages/channels/consumer.py", line 73, in dispatch
django-1  |     await handler(message)
django-1  |   File "/usr/local/lib/python3.12/site-packages/channels/generic/websocket.py", line 173, in websocket_connect
django-1  |     await self.connect()
django-1  |   File "/app/config/consumers.py", line 38, in connect
django-1  |     msg = await sync_to_async(element.websocketmessage)()
django-1  |                               ^^^^^^^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/django/db/models/fields/related_descriptors.py", line 236, in __get__
django-1  |     rel_obj = self.get_object(instance)
django-1  |               ^^^^^^^^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/django/db/models/fields/related_descriptors.py", line 199, in get_object
django-1  |     return qs.get(self.field.get_reverse_related_filter(instance))
django-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/django/db/models/query.py", line 633, in get
django-1  |     num = len(clone)
django-1  |           ^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/django/db/models/query.py", line 380, in __len__
django-1  |     self._fetch_all()
django-1  |   File "/usr/local/lib/python3.12/site-packages/django/db/models/query.py", line 1881, in _fetch_all
django-1  |     self._result_cache = list(self._iterable_class(self))
django-1  |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/django/db/models/query.py", line 91, in __iter__
django-1  |     results = compiler.execute_sql(
django-1  |               ^^^^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/django/db/models/sql/compiler.py", line 1560, in execute_sql
django-1  |     cursor = self.connection.cursor()
django-1  |              ^^^^^^^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/django/utils/asyncio.py", line 24, in inner
django-1  |     raise SynchronousOnlyOperation(message)
django-1  | django.core.exceptions.SynchronousOnlyOperation: You cannot call this from an async context - use a thread or sync_to_async.
django-1  | ERROR:    closing handshake failed
```

File: scram/route_manager/admin.py (hunk 3 above)
Comment (samoehlert) [reply scram/route_manager/admin.py:21]: That sounds like a useful change, thanks.

File: scram/route_manager/models.py (hunk 1 above)
Comment (samoehlert) [reply scram/route_manager/models.py]: Updated to change to that.

File: config/consumers.py (hunk 2 above)
Comment (grigorescu) [reply config/consumers.py:38]: Yuck, ok, thanks.

File: scram/route_manager/models.py (hunk 1 above)
Comment (grigorescu) [reply scram/route_manager/models.py]: I'm not seeing this update?

Comment (samoehlert) [reply scram/route_manager/models.py]: Helps if I actually push..

**Review by grigorescu — APPROVED** [review]

**Pull request merged**